xbmc/cores/dvdplayer/DVDPlayerVideo.cpp

```diff
@@ -811,7 +811,7 @@ int CDVDPlayerVideo::GetLevel() const
     int datasize = m_messageQueue.GetDataSize();
     if (m_pVideoCodec)
       datasize += m_pVideoCodec->GetDataSize();
-    return min(100, (int)(100 * datasize / (m_messageQueue.GetMaxDataSize() * m_messageQueue.GetMaxTimeSize())));
+    return min(100, (int)((100.0 * datasize) / m_messageQueue.GetMaxDataSize()));
```

Comment:
If you're going to turn this into floating point calc, you should probably keep the layout of the time based one and use the round() routine.